app/controllers/concerns/redirect_behavior.rb

```diff
@@ -55,7 +55,7 @@ def redirect_after_submit
       end
     else
       # redirect to feedback page if score is less than 100 percent
-      if @exercise.needs_more_feedback?(@submission) && !@embed_options[:disable_redirect_to_feedback]
+      if @submission.exercise.needs_more_feedback?(@submission) && !@embed_options[:disable_redirect_to_feedback]
```

Comment:
I assume this concern is used in more places. Do we have the `@submission` available in each and every place?
This looks brittle.

Reply:
As I understand it, it is ensured that we have a `@submission` by executing `before_action :set_submission, only: %i[redirect_after_submit]` in `submissions_controller.rb`. All other functions in this file are called only in the `redirect_after_submit` function. However, the problem is that rubocop doesn't like `redirect_after_submit` being used in the `submission_controller.rb` because it's not defined there. Since the `RedirectBehavior` is a module, I can't use `before_action` there. Is there any way to solve this? Because `@submission` is needed for the `redirect_after_submit` function. Copying everything from `module RedirectBehavior` in the `submissions_controller.rb` does not seem to be a good solution.

Reply:
Thanks, Dominic, for paying attention. Indeed, we have the concern in more places -- it's also used by the `CommunitySolutionsController`. There, we are also setting the `@submission` variable, so that it can be used safely.